core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1144,6 +1098,10 @@ public static MetaData fromXContent(XContentParser parser) throws IOException {
                         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
                             builder.put(IndexTemplateMetaData.Builder.fromXContent(parser, parser.currentName()));
                         }
+                    } else if ("warmers".equals(currentFieldName)) {
+                        // TODO: remove this condition in 4.0
```

Comment:
actually this won't help since we load old indices as dangling indices, so warmers won't be read since they are not part of the index metadata. So I guess we need a dedicated test?